codec-http/src/main/java/io/netty/handler/codec/http/HttpContentCompressor.java

```diff
@@ -103,11 +124,75 @@ public HttpContentCompressor(int compressionLevel, int windowBits, int memLevel)
      *        body exceeds the threshold. The value should be a non negative
      *        number. {@code 0} will enable compression for all responses.
      */
+    @Deprecated
     public HttpContentCompressor(int compressionLevel, int windowBits, int memLevel, int contentSizeThreshold) {
         this.compressionLevel = ObjectUtil.checkInRange(compressionLevel, 0, 9, "compressionLevel");
         this.windowBits = ObjectUtil.checkInRange(windowBits, 9, 15, "windowBits");
         this.memLevel = ObjectUtil.checkInRange(memLevel, 1, 9, "memLevel");
         this.contentSizeThreshold = ObjectUtil.checkPositiveOrZero(contentSizeThreshold, "contentSizeThreshold");
+        supportsCompressionOptions = false;
+    }
+
+    /**
+     * Create a new {@link HttpContentCompressor} Instance with specified
+     * {@link CompressionOptions}s and ContentSizeThreshold set to {@code 0}
+     *
+     * @param compressionOptions {@link CompressionOptions} Instance(s)
+     */
+    public HttpContentCompressor(CompressionOptions... compressionOptions) {
+        this(0, compressionOptions);
+    }
+
+    /**
+     * Create a new {@link HttpContentCompressor} Instance with specified
+     * {@link CompressionOptions}s
+     *
+     * @param compressionOptions {@link CompressionOptions} Instance(s)
+     */
+    public HttpContentCompressor(int contentSizeThreshold, CompressionOptions... compressionOptions) {
+        this(Arrays.asList(ObjectUtil.checkNotNull(compressionOptions, "CompressionOptions")));
+    }
+
+    /**
+     * Create a new {@link HttpContentCompressor} Instance with specified
+     * {@link CompressionOptions}s and ContentSizeThreshold set to {@code 0}
+     *
+     * @param compressionOptionsCollection {@link Collection<CompressionOptions>} Instance
+     */
+    public HttpContentCompressor(Collection<CompressionOptions> compressionOptionsCollection) {
```

Comment:
```suggestion
    public HttpContentCompressor(Iterable<CompressionOptions> compressionOptionsCollection) {
```

Reply:
Well the problem with `Iterable` is, I can't use `ObjectUtil.checkNonEmpty(,,,)`.

Reply:
How about adding a variant of `checkNonEmpty` that accepts an `Iterable` then? :wink:

Reply:
Doing that right now. :D